core/src/SYCL/Kokkos_SYCL_Parallel_Reduce.hpp

```diff
@@ -175,27 +175,26 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
         ((size + values_per_thread - 1) / values_per_thread + wgroup_size - 1) /
             wgroup_size,
         1);
-    const auto value_count =
+    const unsigned int value_count =
         FunctorValueTraits<ReducerTypeFwd, WorkTagFwd>::value_count(
             selected_reducer);
-    const auto results_ptr =
-        static_cast<pointer_type>(Experimental::SYCLSharedUSMSpace().allocate(
-            "SYCL parallel_reduce result storage",
-            sizeof(value_type) * std::max(value_count, 1u) * init_size));
+    const auto results_ptr = static_cast<pointer_type>(sycl::malloc_shared(
+        sizeof(*m_result_ptr) * std::max(value_count, 1u) * init_size, q));
```

Comment:
What was wrong with `sizeof(value_type)`?

Reply:
Should work as well.